Решение 2 задач первого спринта

amorozov opened the pull request:
В PR предлагается решение двух задач первого спринта

amorozov committed:
FIXME убран %PUBLIC_URL%

... не могу победить ни одним из способов,
описанных https://bobbyhadz.com/blog/create-react-app-set-public-url

amorozov committed:
frontend -> frontend/microfrontend/microfrontend-shell

Проведена адаптация приложения к использованию в качестве
"основного" микрофронтенда (shell). Убраны лишние
зависимости и команды npm.

TODO: Существующие тесты оставлены в родительском
проекте. В нынешнем виде они вызывают вопросы.

Dockerfile-ы перенесены в соответствующие проекты.
Dockerfile для МФ shell доработан так, чтобы быстрее
собираться и занимать поменьше места.
Контейнер для тестов собирается при помощи Dockerfile.test
и нуждается в доработке.

compose.yaml поправлен соответственно.

Скрипты запуска `npm run-scripts` изменены так, чтобы
запускать соответствующую команду в каждом из микрофронтендов.

amorozov committed:
Переносим в shared-library CurrentUserContext, PopupWithForm и соответствующие CSS.

Эти компоненты потом будут переиспользоваться в других МФ.
Название у библиотеки дурацкое, но лучше нет. В жизни
разделяемых компонентов будет, вероятно, больше и они
будут более логически структурированы, а здесь пока так.

amorozov committed:
Выделен microfrontend-profiles для профилей

Пояснения в microfrontend/microfrontend-profiles/Readme.md

amorozov committed:
Работа с карточками мест вынесена в отдельный MF cards

Пояснения в microfrontend/microfrontend-cards/Readme.md

amorozov committed:
Перенос функционала, связанного с регистрацией и логином пользователей в MF auth

Пояснения в microfrontend/microfrontend-auth/Readme.md

amorozov committed:
Сдача решения предлагает сразу обе задачи

ivannizh commented:
Не смог открыть диаграмму. Можно ее положить в корень репозитория.
![image](https://github.com/user-attachments/assets/a042877f-1f92-4985-b6ed-e81c17b1e6b5)

amorozov committed:
Обновил webpack-cli до последней версии

amorozov committed:
Исправил название события

Изменение потерялось при ребейзе.

amorozov committed:
Добавил саму схему вместо ссылки на неё.

Стыдно признаться, но ссылка, которую даёт Я.Диск,
оказывается нерабочей. Я не знаю, в чём прикол.

amorozov committed:
Удалил зависимость контейнеров фронта от контейнера бэкенда

Вообще, странная зависимость. Ну, нет бэкенда по каким-то
причинам, что теперь, не запускаться вовсе?

amorozov committed:
Заменил команду проверки здоровья контейнера

В используемой версии контейнера nginx
(nginxinc/nginx-unprivileged:1-alpine-slim)
нет curl, поэтому для проверок успешного
запуска nginx использую nc и printf.

amorozov committed:
FIXME: switch to development build type in Docker images

webpack gives slightly broken behavior for production builds

amorozov committed:
Упростил сборку образов

... почти без ухудшения качества.

NPM — ну, вы поняли...

amorozov commented:
Обновил PR:

* исправил название события (потерял коммит при ребейзе)
* понял, почему у Вас не получалось собрать образы, упростил и исправил схему сборки, теперь должно заработать
* адаптировать helathcheck для минимального образа nginx
* добавил в PR саму схему drawio вместо ссылки на неё — так и не понял, что не так с Я.Диском, но разбираться неохота, я им почти не пользуюсь.

amorozov commented:
![изображение](https://github.com/user-attachments/assets/fd6c2f3d-4953-4243-a16c-19fd91217063)

Файл как бы есть, ссылку даёт, я скачать даже я сам его не могу. Извините, что так получилось
